src/ngraph_utils.cc

```diff
@@ -242,6 +242,26 @@ void NgraphSerialize(const std::string& file_name,
                    << std::endl;
     NGRAPH_VLOG(0) << e.what() << std::endl;
   }
+}
+
+void MemoryProfile(long& vm_usage, long& resident_set) {
+  vm_usage = 0;
+  resident_set = 0;
+
+  //Get the two fields we want
+  long vsize;
+  long rss;
+
+  std::ifstream ifs("/proc/self/stat", std::ios_base::in);
+  std::string mem_in; 
+  getline(ifs, mem_in);
+  vector<string> mem_str = ng::split(mem_in, ' ');
+  vsize = std::stol(mem_str[22]);
+  rss = std::stol(mem_str[23]);
```

Comment:
safe to use `stol` here, because `/proc/self/stat` will conceivably not use the `A_POTATO` attack